### DISCUSSION TITLE
Window dragging

### DISCUSSION THREAD

**Discussion by schell** [rust-windowing/winit#1949] [Q&A] [answered]
Hi all. First off let me say that I am enjoying `winit` and I'm very thankful for all your hard work. 

I'm new to `winit` and I am on `aarch64-apple-darwin`. Can we discuss the state of dragging windows by the window bar? It seems that this isn't something that happens automatically and in fact I'm having a difficult time getting it to work at all. I would guess this would be a standard window treatment at least on macos. What's the standard process for enabling window dragging and what are some of the gotchas/caveats? 

Thoughts?

**Comment by maroider** [comment, marked as answer]
> dragging windows by the window bar

I was under the impression that this is something that's supported by macOS, but I don't own a macOS device, so ...

cc @ArturKovacs @madsmtm

**Reply by madsmtm** [reply]
Am using `x86_64-apple-darwin` / MacOS 10.14.6 Mojave myself, but window dragging is definitely something we are supporting

**Comment by madsmtm** [comment]
Can you maybe give some information about your system, as well as try running the `window_debug` example, and see if you can get it to work there / get any errors?

**Comment by schell** [comment]
So it looks like I might have jumped the gun here. I have been launching my window from a terminal that is displayed fullscreen. When the window pops up I can't drag it at all. If I un-fullscreen the terminal window then the `winit` window that was launched becomes draggable. 

So it seems that this is just a very special case where dragging doesn't work if the window is launched from a fullscreen terminal.

**Reply by madsmtm** [reply]
Ah, yeah, I can reproduce that - I've created a new issue to track it here: https://github.com/rust-windowing/winit/issues/1950
